lib/spack/spack/schema/container.py

```diff
@@ -16,35 +53,22 @@
         },
         # Describes the base image to start from and the version
         # of Spack to be used
-        'base': {
-            'type': 'object',
-            'additionalProperties': False,
-            'properties': {
-                'image': {
-                    'type': 'string',
-                    'enum': ['ubuntu:18.04',
-                             'ubuntu:16.04',
-                             'centos:7',
-                             'centos:6']
-                },
-                'spack': {
-                    'type': 'string',
-                    'enum': ['develop', '0.14', '0.14.0']
-                }
-            },
-            'required': ['image', 'spack']
-        },
+        'images': {'anyOf': [_stages_from_dockerhub, _custom_stages]},
         # Whether or not to strip installed binaries
         'strip': {
             'type': 'boolean',
             'default': True
         },
         # Additional system packages that are needed at runtime
         'os_packages': {
-            'type': 'array',
-            'items': {
-                'type': 'string'
-            }
+            'type': 'object',
+            'properties': {
+                'command': {'type': 'string', 'enum': ['apt', 'yum']},
```

Comment:
Each package manager has ways to update the list of packages that can be installed, install them and remove all the data that is not anymore needed once the installation finishes. This is constrained because Spack currently knows only how to deal with `apt` and `yum` (but can be extended later).

Reply:
I'm going to mark this as resolved but to clarify: my suggestion is to let the user specify the full setup/teardown commands e.g. `apt-get install x y z` instead of specifying `apt` and `[x, y, z]`; the former would allow users more flexibility in setting up their container prior to the Spack command.

Reply:
That's a good point, and we may need to adjust it later based on user feedback. The current state is less flexible but also less verbose than what you propose. If the lack of flexibility becomes a problem for users we should definitely change it though.